### DISCUSSION TITLE
包版本指定的疑问

### DISCUSSION THREAD

**Discussion by Extra-Creativity** [xmake-io/xmake#5123] [Q&A]
现在add_requires的版本控制好像是要求用户安装给定范围里的最新版，有没有方式让用户只要存在范围内的版本就允许运行？

**Comment by star-hengxing** [comment]
https://xmake.io/#/package/remote_package?id=semantic-version-settings

**Reply by Extra-Creativity** [reply]
我说的就是这个，比如`1.3.x`会自动检测`1.3.x`中的最新版本（比如`1.3.5`）而不是匹配已经安装的非最新的1.3.x（比如`1.3.4`）

**Reply by star-hengxing** [reply]
默认就是拉取最新版本，你想用哪个，就指定 1.3.4。不过我感觉你想要的是 package lock
https://xmake.io/#/guide/build_policies?id=packagerequires_lock
